src/expr.rs

```diff
@@ -315,12 +305,12 @@ pub fn format_expr(
         })
 }
 
+pub struct PairParts<'a>(pub &'a str, pub &'a str, pub &'a str);
```

Comment:
Could you make this a proper struct rather than tuple struct so that we have some 'documentation' of the fields please? Then we can avoid the destructuring at the top of the function too.